src/main/java/bio/terra/service/profile/ProfileService.java

```diff
@@ -115,11 +118,21 @@ public String updateProfile(BillingProfileUpdateModel billingProfileRequest,
      */
     public String deleteProfile(UUID id, AuthenticatedUserRequest user) {
         iamService.verifyAuthorization(user, IamResourceType.SPEND_PROFILE, id.toString(), IamAction.DELETE);
+        CloudPlatform platform;
+        try {
+            BillingProfileModel billingProfile = profileDao.getBillingProfileById(id);
+            platform = billingProfile.getCloudPlatform();
+        } catch (ProfileNotFoundException e) {
+            // Use default cloud profile if the billing profile does not exist and let the flight handle the object
+            // not being found
+            platform = CloudPlatformWrapper.DEFAULT;
+        }
 
         String description = String.format("Delete billing profile id '%s'", id);
         return jobService
             .newJob(description, ProfileDeleteFlight.class, null, user)
             .addParameter(ProfileMapKeys.PROFILE_ID, id)
+            .addParameter(JobMapKeys.CLOUD_PLATFORM.getKeyName(), platform)
```

Comment:
I wonder if it would work better to use `CloudPlatformWrapper` here instead of `CloudPlatform`.

Reply:
I was wondering about this.  These get serialized into the DB so I like the idea of keeping this smaller